python/foglamp/common/process.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""Common FoglampProcess Class"""
+
+from foglamp.common.storage_client.storage_client import ReadingsStorageClient, StorageClient
+from abc import ABC, abstractmethod
+import argparse
+import http.client
+import json
+
+__author__ = "Ashwin Gopalakrishnan"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+class ArgumentParserError(Exception):
+    """ Overwrite default exception to not terminate application """
+    pass
+
+class FoglampProcess(ABC):
+    """ FoglampProcess for all non-core python processes.
+	All processes will inherit from FoglampProcess and must implement pure virtual method run()
+    """
+
+    _core_management_host = None
+    """ string containing core's microservice management host """
+
+    _core_management_port = None
+    """ int containing core's microservice management port """
+
+    _name = None
+    """ name of process """
+
+    _m_client = None
+    """ MicroserviceManagementClient instance """
+
+    _readings_storage = None
+    """ foglamp.common.storage_client.storage_client.ReadingsStorageClient """
+
+    _storage = None
+    """ foglamp.common.storage_client.storage_client.StorageClient """
+
+    def __init__(self):
+        """
+	    All processes must have these three command line arguments passed:
+		--address [core microservice management host]
+		--port [core microservice management port]
+		--name [process name]
+        """
+        
+        try:    
+            self._core_management_host = self.get_arg_value("--address")
+            self._core_management_port = self.get_arg_value("--port")
+            self._name = self.get_arg_value("--name")
+        except ArgumentParserError:
+            raise
+        if self._core_management_host is None:
+            raise ValueError("--address is not specified")
+        elif self._core_management_port is None:
+            raise ValueError("--port is not specified")
+        elif self._name is None:
+            raise ValueError("--name is not specified")
+
+        self._m_client = self.MicroserviceManagementClient(self._core_management_host,self._core_management_port)
+        self._readings_storage = ReadingsStorageClient(self._core_management_host, self._core_management_port)
+        self._storage = StorageClient(self._core_management_host, self._core_management_port)
+
+    # pure virtual method run() to be implemented by child class
+    @abstractmethod
+    def run(self):
```

Comment:
@ashwinscale  will not this collide with configuration change callback? (Rr we are going to change that to `reconfigure`

Reply:
I don't think it would collide. A callback's run() method would exist in it's own module. The code is basically:
...
cb = import_module(callback)
...
cb.run(category_name)
...